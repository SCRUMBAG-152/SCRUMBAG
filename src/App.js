import React, {
  Component
} from "react";
import Column from "./Component/Column"
import "./App.css";

const DIRECTION_LEFT = -1;
const DIRECTION_RIGHT = 1;

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      columns: [{
          name: "Backlog",
          cards: [{
            name: "Card A"
          }]
        },
        {
          name: "Doing",
          cards: [{
            name: "Card B"
          }]
        },
        {
          name: "Done",
          cards: [{
            name: "Card C"
          }]
        }
      ]
    };
  }

  handleAdd = columnIndex => {
    const name = window.prompt('Name?')
    if (!name) return
    const card = {
      name
    }
    this.setState(prevState => {
      const {
        columns
      } = this.state
      columns[columnIndex].cards.push(card)
      return {
        columns
      }
    })
  }

  handleMove = (columnIndex, cardIndex, direction) => {
    this.setState(prevState => {
      const {
        columns
      } = prevState
      const [card] = columns[columnIndex].cards.splice(cardIndex, 1)
      columns[columnIndex + direction].cards.push(card)
      return {
        columns
      }
    })
  }

  render() {
    return ( <
      div className = "App" > {
        this.state.columns.map((column, columnIndex) => ( <
          Column column = {
            column
          }
          columnIndex = {
            columnIndex
          }
          key = {
            columnIndex
          }
          onMoveLeft = {
            cardIndex => this.handleMove(columnIndex, cardIndex, DIRECTION_LEFT)
          }
          onMoveRight = {
            cardIndex => this.handleMove(columnIndex, cardIndex, DIRECTION_RIGHT)
          }
          onAddCard = {
            () => this.handleAdd(columnIndex)
          }
          />
        ))
      } <
      /div>
    );
  }
}

export default App;