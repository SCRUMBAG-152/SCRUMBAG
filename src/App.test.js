import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('MyComponent', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
  
    expect(component).toMatchSnapshot();
  });
});