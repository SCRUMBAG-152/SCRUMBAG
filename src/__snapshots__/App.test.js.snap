// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    className="App"
  >
    <Switch>
      <Route
        component={[Function]}
        name="Pages"
        path="/pages"
      />
      <Route
        component={[Function]}
        name="Home"
        path="/"
      />
    </Switch>
  </div>
</Router>
`;
